frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -300,18 +303,16 @@ function QueryBuilder(props) {
     return () => window.removeEventListener("resize", forceUpdateDebounced);
   });
 
-  const isExistingModelDirty = useMemo(
-    () => isModelQueryDirty || isMetadataDirty,
-    [isMetadataDirty, isModelQueryDirty],
-  );
+  const shouldShowUnsavedChangesWarningForModels =
```

Comment:
Wdyt about extracting this logic in a separate hook (in this file) and use `useSelector` instead of adding more props to the root component?

Reply:
sounds good to me.